README.md

```diff
@@ -221,6 +221,48 @@ The const array prototype is a const object that contains the same methods as Ar
 - `ConstArray.prototype.pop()` and `ConstArray.prototype.shift()` do not return the removed element, they return the result of the change
 - `ConstArray.prototype.first()` and `ConstArray.prototype.last()` are added to return the first and last element of the const array
 
+## Usage in {`Map`|`Set`|`WeakMap`}
+
+It is possible to use a const object or const array as a key in a `Map`, and as a value in a `Set`. When using a const object or const array in this way, key/value equality behaves as expected.
+
+It is not possible to use a const object or const array as a key in a `WeakMap`, because const objects and const arrays are not `Objects`, and their lifetime is not observable. Attempting to set a value in a WeakMap using a const object
```

Comment:
this is something i'll bring up in committee when the time arises; these are certainly objects if `Object(x) === x`, and `typeof x` is `'object'` or `'function'`.

Reply:
`Object(x) !== x`, because `Object(x)` returns a mutable object.

`typeof x` will be `constobject` or `record` or whatever thing we need to decide on (i think that @rickbutton is working on an another PR to clarify that)

Reply:
And I did not check but #23 is that PR!

Reply:
no, it does not return a mutable object - `Object(Object.freeze(x))` for Object x returns a frozen object.

Reply:
exactly, `Object(x)` will copy everything into a new mutable/regular/`typeof z === "object"` object.

Reply:
in that case, it sounds like this proposal defines new primitives, so “const object” would be a misleading name.

Reply:
We do not have the same semantics as `Object.freeze` here. Yes we'd copy everything to the new mutable object

Reply:
> in that case, it sounds like this proposal defines new primitives, so “const object” would be a misleading name (i see the proposal is named const value types, so i must have gotten confused seeing that verbiage somewhere)

Yes we're kind of trying to find a new name instead of const value types. I think I'll open an issue soon about it